### ISSUE TITLE
public property inside anything other than class generates AD0001: 'System.NullReferenceException' 

### ISSUE THREAD

**Issue by echoix** [d1820/CodeDocumentor#8]
The analyzer 'CodeDocumentor.PropertyAnalyzer' has problems when properties aren't inside a class. I encounter it with my interfaces, and this repo encounters it with structs.

I see in the following file multiple places where they cast the `node.Parent` as `ClassDeclarationSyntax`, but it isn't always the case. https://github.com/d1820/CodeDocumentor/blob/c51ff74b2eb3a731a8d890bdf8de3d5d7424b675/CodeDocumentor/Helper/PrivateMemberVerifier.cs#L54-L56

```
Analyzer 'CodeDocumentor.PropertyAnalyzer' threw an exception of type 'System.NullReferenceException' with message 'La référence d'objet n'est pas définie à une instance d'un objet.'.
Exception occurred with following context:
Compilation: CodeDocumentor.Test
SyntaxTree: D:\repos\CodeDocumentor\CodeDocumentor.Test\TestHelpers\DiagnosticResult.cs
SyntaxNode: public int Line { get; } [PropertyDeclarationSyntax]@[918..942) (31,8)-(31,32)

System.NullReferenceException: La référence d'objet n'est pas définie à une instance d'un objet.
   à CodeDocumentor.Helper.PrivateMemberVerifier.IsPrivateMember(ClassDeclarationSyntax node) dans C:\Dev\Personal\CodeDocumentor\CodeDocumentor\Helper\PrivateMemberVerifier.cs:ligne 19
   à CodeDocumentor.Helper.PrivateMemberVerifier.IsPrivateMember(PropertyDeclarationSyntax node) dans C:\Dev\Personal\CodeDocumentor\CodeDocumentor\Helper\PrivateMemberVerifier.cs:ligne 85
   à CodeDocumentor.PropertyAnalyzer.AnalyzeNode(SyntaxNodeAnalysisContext context) dans C:\Dev\Personal\CodeDocumentor\CodeDocumentor\Analyzers\Properties\PropertyAnalyzer.cs:ligne 45
   à Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.<>c__62`1.<ExecuteSyntaxNodeAction>b__62_0(ValueTuple`2 data)
   à Microsoft.CodeAnalysis.Diagnostics.AnalyzerExecutor.ExecuteAndCatchIfThrows_NoLock[TArg](DiagnosticAnalyzer analyzer, Action`1 analyze, TArg argument, Nullable`1 info)
-----

Suppress the following diagnostics to disable this analyzer: CD1606
```

I made a simple test case for PropertyUnitTests.cs for an interface to reproduce one of these problems.

```patch
diff --git a/CodeDocumentor.Test/PropertyUnitTests.cs b/CodeDocumentor.Test/PropertyUnitTests.cs
index 82c737f..ade6328 100644
--- a/CodeDocumentor.Test/PropertyUnitTests.cs
+++ b/CodeDocumentor.Test/PropertyUnitTests.cs
@@ -62,6 +62,41 @@ namespace ConsoleApp4
         public string PersonName { get; set; }
 	}
 }";
+      
+        /// <summary>
+        /// The property with getter setter in public interface test code.
+        /// </summary>
+        private const string PropertyWithGetterSetterInPublicInterfaceTestCode = @"
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ConsoleApp4
+{
+	public interface IPropertyTester
+	{
+		public string PersonName { get; set; }
+	}
+}";
+
+        /// <summary>
+        /// The property with getter setter in public interface test fix code.
+        /// </summary>
+        private const string PropertyWithGetterSetterPublicInterfaceTestFixCode = @"
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ConsoleApp4
+{
+	public interface IPropertyTester
+	{
+        /// <summary>
+        /// Gets or Sets the person name.
+        /// </summary>
+        public string PersonName { get; set; }
+	}
+}";
 
         /// <summary>
         /// The property only getter test code.
@@ -345,6 +380,7 @@ namespace ConsoleApp4
         /// <param name="column">The column.</param>
         [Theory]
         [InlineData(PropertyWithGetterSetterTestCode, PropertyWithGetterSetterTestFixCode, 10, 17)]
+        [InlineData(PropertyWithGetterSetterInPublicInterfaceTestCode, PropertyWithGetterSetterPublicInterfaceTestFixCode, 10, 17)]
         [InlineData(PropertyOnlyGetterTestCode, PropertyOnlyGetterTestFixCode, 10, 17)]
         [InlineData(PropertyPrivateGetterTestCode, PropertyPrivateGetterTestFixCode, 10, 23)]
         [InlineData(PropertyInternalGetterTestCode, PropertyInternalGetterTestFixCode, 10, 23)]

```

I'm not experienced enough in analyzers to help very much. But I saw that the code seems to run with .net Framework 4.7.2 (is it normal?), which I don't have much experience with, however it is using multiple time `as` to cast, without checking if the cast returned null, nor is using nullable reference types, that seems to be a more modern language construct.

**Comment by d1820** [comment]
Resolved with v 1.1.4

**Comment by echoix** [comment]
Thanks!
